libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -2278,23 +2278,36 @@ void ExpressionCompiler::appendAndOrOperatorCode(BinaryOperation const& _binaryO
 
 void ExpressionCompiler::appendCompareOperatorCode(Token _operator, Type const& _type)
 {
-	solAssert(_type.sizeOnStack() == 1, "Comparison of multi-slot types.");
+	// solAssert(_type.sizeOnStack() == 1, "Comparison of multi-slot types.");
+	// "Might still be useful to keep this assert, everywhere except for external function types." --Comment from issue
+	// TO DO :: check for the implications of this comment. And see if and where to we retain this solAssert
+
 	if (_operator == Token::Equal || _operator == Token::NotEqual)
 	{
-		if (FunctionType const* funType = dynamic_cast<decltype(funType)>(&_type))
+		if (_operator == Token::Equal || _operator == Token::NotEqual)
```

Comment:
This ``if`` is redundant, isn't it?
Apart from that, it looks better now, yes.

Reply:
Indeed yes, it's redundant, I've removed it in the latest push. 
Thanks for pointing it out!